Simplification

chabbimilind opened the pull request.

chabbimilind committed:
Rename makicLock to optiLock.

chabbimilind committed:
simplified deeply nested if/else statements.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber-research/GOCC?pullRequest=2) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/uber-research/GOCC?pullRequest=2) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: lollllcat<br/>:x: chabbimilind<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber-research/GOCC?pullRequest=2) it.</sub>

chabbimilind committed:
[bugfix]: added missing continue statement in a conditional.

chabbimilind committed:
[bugfix] do not combime Call and Defer in the same case.

Call and Defer are different types, hence we cannot type cast Defer to a Call.
Instead, use different case statements to extract the Call.Value present in both of them,
and then we can reuse the rest of the code on the Value type.

lollllcat committed:
Merge branch 'DIP' into simplification

Commit:
fix bug on instruction type
